### ISSUE TITLE
php-ide crash

### ISSUE THREAD

**Issue by Kaczuc** [atom/ide-php#160]
When he wants to edit a file with the extension php he gets this message

![image](https://user-images.githubusercontent.com/43011737/102496457-b365a100-4077-11eb-844c-ee8266b59448.png)

settings:
![image](https://user-images.githubusercontent.com/43011737/102496540-d001d900-4077-11eb-9a2f-32f3ed9cc529.png)

**Comment by Kaczuc** [comment]
New error
![image](https://user-images.githubusercontent.com/43011737/102497618-27547900-4079-11eb-9caf-21a35be26894.png)

**Comment by donaldinou** [comment]
Related to the package ```netresearch/jsonmapper``` as a dependency from ```felixfbecker/language-server```. 
It is required as follow: ```"netresearch/jsonmapper": "^1.0"``` .
Regarding to the **composer.lock** it has been released with the v1.4.0 but many fixes on 1.x branch exists now (last release is 1.6.0)

So the solution is to do a ```composer update``` or force >=1.6.0 on the root package and then ```composer install```.
Finally, republish the plugin.

**Comment by ErvinSabic** [comment]
Can confirm this bug. Happened right after I updated to PHP 8.
![image](https://user-images.githubusercontent.com/23107956/106225341-8d6d0980-61aa-11eb-92ae-2725a02ab188.png)

**Comment by Ritotsume** [comment]
> Can confirm this bug. Happened right after I updated to PHP 8.
> ![image](https://user-images.githubusercontent.com/23107956/106225341-8d6d0980-61aa-11eb-92ae-2725a02ab188.png)

I fixed this in this way:

Open JsonMapper.php (I use Archlinux):
`$HOME/.atom/packages/ide-php/vendor/netresearch/jsonmapper/src/JsonMapper.php`

And change this on line 300:
```
// Turn this...
if ($type !== '' && $type{0} != '\\') {

// Into this...
if ($type !== '' && $type[0] != '\\') {
```
Restart Atom and the error is gone. For me worked. I hope this help you.

**Comment by donaldinou** [comment]
> > Can confirm this bug. Happened right after I updated to PHP 8.

You can fix this way but you’re changing an old vendor library. There are other fixes of you just use composer to update the plugin on the « riche way » (composer update)

**Comment by Ritotsume** [comment]
> > > Can confirm this bug. Happened right after I updated to PHP 8.
> 
> You can fix this way but you’re changing an old vendor library. There are other fixes of you just use composer to update the plugin on the « riche way » (composer update)

Yes, you're right. That was only a workaround, becouse I got a lot of other errors related to PHP version.

Again, you're right. For the questions of time, I used that workaround. Later, I will use the right way.

**Comment by LoganTann** [comment]
I still have the same error.
I'm pasting it for SEO : 
```
DEBUG The xdebug extension is not loaded
DEBUG Listening on STDIN
PHP Fatal error: Array and string offset access syntax with curly braces is no longer supported in /home/logan/.atom/packages/ide-php/vendor/netresearch/jsonmapper/src/JsonMapper.php on line 300
PHP Fatal error: Array and string offset access syntax with curly braces is no longer supported in /home/logan/.atom/packages/ide-php/vendor/netresearch/jsonmapper/src/JsonMapper.php on line 300
```

**Comment by cricketmon** [comment]
My solution was this after upgrading to php@8:

-Visit https://github.com/cweiske/jsonmapper
-Go to src/JsonMapper.php ( there u will see a 'Add support for PHP 8.0' banner =) )
-Copy the entire getFullNamespace (line 329 to 341)

Till the pull request its done officially this will do the job perfectly
[#158](https://github.com/atom/ide-php/pull/158)
